akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1553,8 +1553,14 @@ final class Delay[T](val d: FiniteDuration, val strategy: DelayOverflowStrategy)
       completeIfReady()
 
     def onPull(): Unit = {
-      if (!isTimerActive(timerName) && !buffer.isEmpty && nextElementWaitTime() < 0)
-        push(out, buffer.dequeue()._2)
+      if (!isTimerActive(timerName) && !buffer.isEmpty) {
+        val waitTime = nextElementWaitTime()
+        if (waitTime < 0) {
+          push(out, buffer.dequeue()._2)
+        } else {
+          scheduleOnce(timerName, Math.max(10, waitTime).millis)
```

Comment:
This is now the second place where the constant `10` is used. Please factor it out to a `final val DelayPrecisionMs = 10`